src/main/java/net/snowflake/client/jdbc/ConnectionHandler.java

```diff
@@ -0,0 +1,42 @@
+package net.snowflake.client.jdbc;
+
+import java.sql.*;
+import java.util.Properties;
+import net.snowflake.client.core.BaseSFSession;
+import net.snowflake.client.core.SFBaseResultSet;
+
+/**
+ * Factory class that presents the implementation of a Snowflake Connection. This allows for
+ * alternate definitions of SFSession, SFStatement, and SFResultSet, (representing the 'physical'
+ * implementation layer) that can share high-level code.
+ */
+public interface ConnectionHandler {
```

Comment:
Should we call it `SFConnectionHandler`. My feeling is that all public interface to customers are using `Snowflake` prefix and internal ones are using `SF` prefix.

Reply:
I'm fine with that.